File: SmartLink/Source/SmartLinkClient/NRC_ServiceGenericClasses/Exception_PathNotFound.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
//This class adds a Path property to the exception class and can be used to
//track invalid directories or file paths..
namespace NRC.Miscellaneous
{

	public class Exception_PathNotFound : Exception
	{


		private string _strPath;
		public string Path {
			get { return _strPath; }
			set { _strPath = value; }
		}

		public Exception_PathNotFound(string Path) : base("An exception occurred when trying access this path: " + Path)
		{
			this._strPath = Path;
		}

		public Exception_PathNotFound(string Path, Exception Inner) : base("An exception occurred when trying access this path: " + Path, Inner)
		{
			this._strPath = Path;
		}

	}

}

File: SmartLink/Source/SmartLinkClient/NRC_ServiceGenericClasses/Exception_PathNotFound.vb
'This class adds a Path property to the exception class and can be used to
'track invalid directories or file paths..
Namespace Miscellaneous

    Public Class Exception_PathNotFound
        Inherits Exception

        Private _strPath As String

        Public Property Path() As String
            Get
                Return _strPath
            End Get
            Set(ByVal value As String)
                _strPath = value
            End Set
        End Property

        Public Sub New(ByVal Path As String)
            MyBase.New("An exception occurred when trying access this path: " & Path)
            Me._strPath = Path
        End Sub

        Public Sub New(ByVal Path As String, ByVal Inner As Exception)
            MyBase.New("An exception occurred when trying access this path: " & Path, Inner)
            Me._strPath = Path
        End Sub

    End Class

End Namespace
